src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/CcdMessageProcessor.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.reform.wacaseeventhandler.services;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.EvaluateResponse;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.EventInformation;
+import uk.gov.hmcts.reform.wacaseeventhandler.handlers.CaseEventHandler;
+
+import java.util.List;
+
+@Slf4j
+@Service
+public class CcdMessageProcessor {
+
+    private final List<CaseEventHandler> handlerServices;
+    private final ObjectMapper objectMapper;
+
+    public CcdMessageProcessor(List<CaseEventHandler> handlerServices, ObjectMapper objectMapper) {
+        this.handlerServices = handlerServices;
+        this.objectMapper = objectMapper;
+    }
+
+    @SuppressWarnings("PMD.DataflowAnomalyAnalysis")
+    public void processMesssage(String message) throws JsonProcessingException {
+        EventInformation eventInformation = objectMapper.readValue(message, EventInformation.class);
+        log.info(String.format("Message received from topic: %s", eventInformation.toString()));
+
+        for (CaseEventHandler handler : handlerServices) {
+            List<? extends EvaluateResponse> results = handler.evaluateDmn(eventInformation);
```

Comment:
What happens when errors are coming from the handlers? Do we send them to that dead letter queue?

Reply:
yes, any application error while processing the message need to be dead letter queued.